test/integration/h2_capture_fuzz.proto

```diff
@@ -0,0 +1,158 @@
+syntax = "proto3";
+
+package test.integration;
+
+message H2FramePing {
+  enum Flags {
+    None = 0;
+    Ack = 1;
+  }
+  Flags flags = 1;
+  bytes data = 2;
+}
+
+message H2FrameSettings {
+  enum Flags {
+    None = 0;
+    Ack = 1;
+  }
+  Flags flags = 1;
+}
+
+enum HeadersFlags {
+  None = 0;
+  EndStream = 1;
+  EndHeaders = 4;
+}
+
+message H2FrameHeaders {
+  HeadersFlags flags = 1;
+  uint32 stream_index = 2;
+}
+
+message H2FrameContinuation {
+  HeadersFlags flags = 1;
+  uint32 stream_index = 2;
+}
+
+message H2FrameData {
+  enum Flags {
+    None = 0;
+    EndStream = 1;
+  }
+  Flags flags = 1;
+  uint32 stream_index = 2;
+}
+
+message H2FramePriority {
+  uint32 stream_index = 1;
+  uint32 dependent_index = 2;
+}
+
+enum H2ErrorCode {
+  NoError = 0;
+  ProtocolError = 1;
+  InternalError = 2;
+  FlowControlError = 3;
+  SettingsTimeout = 4;
+  StreamClosed = 5;
+  FrameSizeError = 6;
+  RefusedStream = 7;
+  Cancel = 8;
+  CompressionError = 9;
+  ConnectError = 10;
+  EnhanceYourCalm = 11;
+  InadequateSecurity = 12;
+  Http11Required = 13;
+}
+
+message H2FramePushPromise {
+  HeadersFlags flags = 1;
+  uint32 stream_index = 2;
+  uint32 promised_stream_index = 3;
+}
+
+message H2FrameResetStream {
+  uint32 stream_index = 1;
+  H2ErrorCode error_code = 2;
+}
+
+message H2FrameGoAway {
+  uint32 last_stream_index = 1;
+  H2ErrorCode error_code = 2;
+}
+
+message H2FrameWindowUpdate {
+  uint32 stream_index = 1;
+  uint32 increment = 2;
+}
+
+// A header that contains invalid status
+message H2FrameMalformedRequest {
+  uint32 stream_index = 1;
+}
+
+// A request that is comprised of a header that has HTTP GET request with a given host and path and
+// an additional zero length header (making this a malformed request)
+message H2FrameMalformedRequestWithZerolenHeader {
+  uint32 stream_index = 1;
+  bytes host = 2;
+  bytes path = 3;
+}
+
+// A request that is comprised of a header that has HTTP GET request with a given host and path
+message H2FrameRequest {
+  uint32 stream_index = 1;
+  bytes host = 2;
+  bytes path = 3;
+}
+
+// A request that is comprised of a header that has HTTP POST request with a given host and path
+message H2FramePostRequest {
+  uint32 stream_index = 1;
+  bytes host = 2;
+  bytes path = 3;
+}
+
+message H2TestFrame {
+  // These values map to the frame creation methods in:
+  // test/common/http/http2/http2_frame.h
+  oneof frame_type {
```

Comment:
This is awesome, I'm excited to see metadata frames to this later. @AndresGuedez

Reply:
Sure. Haven't found a METADATA frame in the [RFC](https://tools.ietf.org/html/rfc7540#section-6), but would be glad to add it if there's a spec.